_ml-commons-plugin/connectors.md

```diff
@@ -0,0 +1,461 @@
+
+```json
+{
+  "task_id": "cVeMb4kBJ1eYAeTMFFgj",
+  "status": "CREATED"
+}
+```
+
+
+You can use the `task_id` to find the `model_id`, as shown the following example:
+
+
+**GET task request**
+
+```json
+GET /_plugins/_ml/tasks/cVeMb4kBJ1eYAeTMFFgj
+```
+
+**GET task response**
+
+```json
+{
+  "model_id": "cleMb4kBJ1eYAeTMFFg4",
+  "task_type": "REGISTER_MODEL",
+  "function_name": "REMOTE",
+  "state": "COMPLETED",
+  "worker_node": [
+    "XPcXLV7RQoi5m8NI_jEOVQ"
+  ],
+  "create_time": 1689793598499,
+  "last_update_time": 1689793598530,
+  "is_async": false
+}
+```
+
+Lastly, use the `model_id` to deploy the model:
+
+**Deploy model request**
+
+```json
+POST /_plugins/_ml/models/cleMb4kBJ1eYAeTMFFg4/_deploy
+```
+
+**Deploy model response**
+
+```json
+{
+  "task_id": "vVePb4kBJ1eYAeTM7ljG",
+  "status": "CREATED"
+}
+```
+
+Use the `task_id` from the deploy model response to make sure the model deployment completes:
+
+**Verify deploy completion request**
+
+```json
+GET /_plugins/_ml/tasks/vVePb4kBJ1eYAeTM7ljG
+```
+
+**Verify deploy completion response**
+
+```
+{
+  "model_id": "cleMb4kBJ1eYAeTMFFg4",
+  "task_type": "DEPLOY_MODEL",
+  "function_name": "REMOTE",
+  "state": "COMPLETED",
+  "worker_node": [
+    "n-72khvBTBi3bnIIR8FTTw"
+  ],
+  "create_time": 1689793851077,
+  "last_update_time": 1689793851101,
+  "is_async": true
+}
+```
+
+After a successful deployment, you can test the model using the Predict API set in the connector's `action` settings, as shown in the following example:
+
+```json
+POST /_plugins/_ml/models/cleMb4kBJ1eYAeTMFFg4/_predict
+{
+  "parameters": {
+    "model": "gpt-3.5-turbo",
+    "messages": [
+      {
+        "role": "system",
+        "content": "You are a helpful assistant."
+      },
+      {
+        "role": "user",
+        "content": "Hello!"
+      }
+    ]
+  }
+}
+```
+
+The Predict API returns inference results for the connected model, as shown in the following example response:
+
+```json
+{
+  "inference_results": [
+    {
+      "output": [
+        {
+          "name": "response",
+          "dataAsMap": {
+            "id": "chatcmpl-7e6s5DYEutmM677UZokF9eH40dIY7",
+            "object": "chat.completion",
+            "created": 1689793889,
+            "model": "gpt-3.5-turbo-0613",
+            "choices": [
+              {
+                "index": 0,
+                "message": {
+                  "role": "assistant",
+                  "content": "Hello! How can I assist you today?"
+                },
+                "finish_reason": "stop"
+              }
+            ],
+            "usage": {
+              "prompt_tokens": 19,
+              "completion_tokens": 9,
+              "total_tokens": 28
+            }
+          }
+        }
+      ]
+    }
+  ]
+}
+```
+
+### Internal connector
+
+To create an internal connector, add the `connector` parameter to the Register model API, as shown in the following example:
+
+```json
+POST /_plugins/_ml/models/_register
+{
```

Comment:
This is not chat model example. Let's keep consistent with the standalone connector.